src/main/java/digital/slovensko/autogram/core/eforms/EFormUtils.java

```diff
@@ -0,0 +1,284 @@
+        var element = xdc.getElementsByTagNameNS("http://data.gov.sk/def/container/xmldatacontainer+xml/1.1", elementLocalName)
+                .item(0);
+        if (element == null)
+            throw new XMLValidationException("Zlyhala validácia XML Datacontainera", "Element " + elementLocalName + " nebol nájdený");
+
+        var attributes = element.getAttributes();
+        if (attributes == null || attributes.getLength() == 0)
+            throw new XMLValidationException("Zlyhala validácia XML Datacontainera", "Atribúty v " + elementLocalName + " neboli nájdené");
+
+        var digestValue = attributes.getNamedItem("DigestValue");
+        if (digestValue == null)
+            throw new XMLValidationException("Zlyhala validácia XML Datacontainera", "Hodnota odtlačku v " + elementLocalName + " nebola nájdená");
+
+        return digestValue.getNodeValue();
+    }
+
+    public static String getFormUri(Element xdc) {
+        var xmlData = xdc.getElementsByTagNameNS(
+                "http://data.gov.sk/def/container/xmldatacontainer+xml/1.1", "XMLData").item(0);
+
+        if (xmlData == null)
+            return null;
+
+        return xmlData.getAttributes().getNamedItem("Identifier").getNodeValue();
+    }
+
+    public static String getNamespaceFromEformXml(Node xml) {
+        var xmlns = xml.getAttributes().getNamedItem("xmlns");
+        if (xmlns == null)
+            return null;
+
+        return xmlns.getNodeValue();
+    }
+
+    public static byte[] getTransformation(String url) {
+        return getResource(url + "/form.sb.xslt");
+    }
+
+    public static byte[] getSchema(String url) {
+        return getResource(url + "/form.xsd");
+    }
+
+    private static byte[] getResource(String url) {
+        var offlineFileLoader = new FileCacheDataLoader();
+        offlineFileLoader.setCacheExpirationTime(21600000);
+        offlineFileLoader.setDataLoader(new CommonsDataLoader());
+
+        DSSDocument xsltDoc;
+        try {
+            xsltDoc = offlineFileLoader.getDocument(url);
+        } catch (DSSException e) {
+            return null;
+        }
+
+        if (xsltDoc == null)
+            return null;
+
+        try {
+            return xsltDoc.openStream().readAllBytes();
+        } catch (IOException e) {
+            return null;
+        }
+    }
+
+    public static Document getXmlFromDocument(DSSDocument documentToDisplay) throws XMLValidationException {
+        try {
+            var builderFactory = DocumentBuilderFactory.newInstance();
+            builderFactory.setNamespaceAware(true);
+            builderFactory.setFeature("http://xml.org/sax/features/external-general-entities", false);
+
+            var is = documentToDisplay.openStream();
+            var inputSource = new InputSource(is);
+            inputSource.setEncoding(ENCODING.displayName());
+            var parsedDocument = builderFactory.newDocumentBuilder().parse(inputSource);
```

Comment:
Toto pouzivame pomerne casto a vyzera, ze to je potencialne derave (ak zabudnes ten setfeature), nedame to do nejakeho utils kde sa to bude dat pouzivat?

Reply:
súhlasím. Znížili som počet výskytov tejto veci, ale môžeme to stiahnuť aj na 1.

Reply:
Spravil som XMLUtils a pre každú z týchto troch Factories som spravil jednu metódu.